src/main/java/frc/robot/subsystems/ArmSubsystem.java

```diff
@@ -110,134 +113,68 @@ public ArmSubsystem() {
         }
 
         // Set default PID values elbow
-        for (var config : new PID[]{ELBOW_DEFAULT, ELBOW_STEADY}) {
+        for (var config : new PID[] {ELBOW_DEFAULT, ELBOW_STEADY}) {
             m_elbow.getPIDController().setP(config.P, config.slot);
             m_elbow.getPIDController().setI(config.I, config.slot);
             m_elbow.getPIDController().setD(config.D, config.slot);
             m_elbow.getPIDController().setFF(config.FF, config.slot);
             m_elbow.getPIDController().setDFilter(config.DF, config.slot);
         }
 
+        m_elbow.enableSoftLimit(SoftLimitDirection.kForward, true);
+        m_elbow.enableSoftLimit(SoftLimitDirection.kReverse, true);
+        m_shoulder.enableSoftLimit(SoftLimitDirection.kForward, true);
+        m_shoulder.enableSoftLimit(SoftLimitDirection.kReverse, true);
+
         // Set the soft limits
         m_shoulder.setSoftLimit(CANSparkMax.SoftLimitDirection.kForward, 60);
         m_shoulder.setSoftLimit(CANSparkMax.SoftLimitDirection.kReverse, -15);
         m_elbow.setSoftLimit(CANSparkMax.SoftLimitDirection.kForward, 60);
         m_elbow.setSoftLimit(CANSparkMax.SoftLimitDirection.kReverse, -15);
+    }
 
+    public void setState(State desiredState) {
+        this.m_desiredState = desiredState;
     }
 
-    public void setState(State state) {
-        // Don't do anything already in correct state
-        if (state == m_state_actual)
-            return;
-
-        switch (m_state_actual) {
-            case Pickup:
-                if (state == State.Rest)
-                    m_state_desired = state;
-                break;
-            case Rest:
-                m_state_desired = state;
-                break;
-            case PlacingA:
-            case PlacingB:
-            case PlacingC:
-            case PlacingD:
-            case PlacingE:
-                m_state_desired = state;
-                break;
-            case Transit:
-                break;
-        }
+    public boolean atState() {
+        boolean shoulder_at_state = goalCalculator(m_shoulder.getEncoder().getPosition(), ArmConstants.SHOULDER_GEAR_RATIO(m_desiredState.shoulder));
+        boolean elbow_at_state = goalCalculator(m_elbow.getEncoder().getPosition(), ArmConstants.ELBOW_GEAR_RATIO(m_desiredState.elbow));
+        return shoulder_at_state && elbow_at_state;
+    }
+
+    private boolean goalCalculator(double current, double desired) {
+        if (current == desired) return true;

```

Comment:
Remove this line. It is redundant.

Reply:
L47? It's an optimization because Math.abs takes a pretty long time. This is just a general method I copied from the web, anyways. Removing it will make it a little slower, is that fine?